spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1207,7 +1223,7 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
                     .newBuilder()
                     .setChild(e)
                     .setDatatype(serializeDataType(IntegerType).get)
-                    .setEvalMode("LEGACY") // year is not affected by ANSI mode
+                    .setEvalMode(stringToEvalMode("LEGACY")) // year is not affected by ANSI mode
```

Comment:
You can just use the enum directly here rather than convert from string.

```suggestion
                    // year is not affected by ANSI mode so we use LEGACY mode here
                    .setEvalMode(ExprOuterClass.EvalMode.LEGACY)
```

Reply:
Thank you for another great suggestion. 
Updated accordingly to `QueryPlanSerde.scala` L1226.